sui_programmability/framework/sources/Coin.move

```diff
@@ -122,18 +129,26 @@ module Sui::Coin {
     }
 
     /// Create a coin worth `value`. and increase the total supply
-    /// in `cap` accordingly
-    /// Aborts if `value` + `cap.total_supply` >= U64_MAX
+    /// in `cap` accordingly.
     public fun mint<T>(
```

Comment:
When would someone ever need to call the `mint` function?

Reply:
Well, yeah, we have overlapping logic here. I imagine that if someone was creating a new Coin and then sending it right away, he'd be able to do it in two ways:

```rs
Transfer::transfer(Coin::mint(..), user1);
Transfer::transfer(Coin::mint(..), user2);
Transfer::transfer(Coin::mint(..), me);
```
OR
```rs
let balance = Coin::mint_balance();

Transfer::transfer(Coin::withdraw(&mut balance, ctx), user1);
Transfer::transfer(Coin::withdraw(&mut balance, ctx), user2);
Transfer::transfer(Coin::from_balance(balance, ctx), me);
```